.github/workflows/cherry-pick-milestoned-prs.yml

```diff
@@ -0,0 +1,406 @@
+              '❌ **Cherry-pick to `' + targetBranch + '` failed.**\n\n' +
+              '**Error:** ' + errorMsg + '\n\n' +
+              '---\n\n' +
+              'IMPORTANT: Merging this PR to the appropriate branches is critical to the release process.\n\n' +
+              '**Workflow run:** ' + workflowLink;
+
+            await github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: parseInt('${{ needs.prepare.outputs.pr_number }}'),
+              body: body
+            });
+
+      # Notify Slack about failed cherry-pick to milestoned branch
+      - name: Notify Slack on failure
+        uses: archive/github-actions-slack@c643e5093620d65506466f2c9b317d5d29a5e517 # v2.10.1
+        env:
+          WORKFLOW_RUN_URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+          SOURCE_PR_URL: ${{ github.event.pull_request.html_url }}
+          SOURCE_PR_NUMBER: ${{ github.event.pull_request.number }}
+          SOURCE_PR_TITLE: ${{ github.event.pull_request.title }}
+          TARGET_BRANCH: ${{ needs.prepare.outputs.milestoned_branch }}
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.CODE_FREEZE_BOT_TOKEN }}
+          slack-channel: ${{ secrets.WOO_RELEASE_SLACK_NOTIFICATION_CHANNEL }}
+          slack-text: |
+            :warning: *Cherry pick from `trunk` to `${{ env.TARGET_BRANCH }}` <${{ env.WORKFLOW_RUN_URL }}|failed>*  
+            Please resolve before releasing.  
+            Source PR: _<${{ env.SOURCE_PR_URL }}|#${{ env.SOURCE_PR_NUMBER }} - ${{ env.SOURCE_PR_TITLE }}>_.
+
+  # Handle failed cherry-pick to next branch
+  handle-failures-next:
+    needs: [prepare, cherry-pick-next]
+    runs-on: ubuntu-latest
+    if: ${{ always() && needs.cherry-pick-next.result == 'failure' }}
+    steps:
+      - name: Handle failure for next branch
+        if: ${{ needs.cherry-pick-next.outputs.status == 'failed' }}
+        run: echo "Processing failure for ${{ needs.prepare.outputs.next_branch }}"
+
+      # Label original PR with failure tag
+      - name: Add "cherry pick failed" label
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.issues.addLabels({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: parseInt('${{ needs.prepare.outputs.pr_number }}'),
+              labels: ['cherry pick failed']
+            })
+
+      # Comment on original PR about failure
+      - name: Comment on original PR
+        uses: actions/github-script@v7
+        env:
+          ERROR_MESSAGE: ${{ needs.cherry-pick-next.outputs.error_message || 'Unknown error' }}
+          TARGET_BRANCH: ${{ needs.prepare.outputs.next_branch }}
+          GITHUB_RUN_ID: ${{ github.run_id }}
+        with:
+          script: |
+            const pr = (await github.rest.pulls.get({
```

Comment:
Should we handle this request in a `try/catch` as we do in the `handle-successes-milestoned`?